mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassin.java

```diff
@@ -29,33 +29,55 @@
 import org.apache.james.util.Host;
 
 import com.github.fge.lambdas.Throwing;
+import com.google.inject.ImplementedBy;
 
-public class SpamAssassin {
+@ImplementedBy(SpamAssassin.SpamAssassinImpl.class)
```

Comment:
you must not use that:
1. it reverts the IoC principle (impl knows interface but not the opposite)
2. it's guava specific (and you should not be able to use guava in general purpose code)